### PR TITLE
make tests for syntax errors not call the binary

### LINKED ISSUE #114: Numerous tests don't need to run entire bin/uglifyjs binary
**Testing improvement**

Numerous tests rely on starting bin/uglifyjs binary just to exercise lib/parse.js:

```
    ✓ Should throw syntax error (5--) (199ms)
    ✓ Should throw syntax error (Math.random() /= 2) (206ms)
    ✓ Should throw syntax error (++this) (202ms)
    ✓ Should throw syntax error (++null) (198ms)
    ✓ Should throw syntax error (a.=) (197ms)
    ✓ Should throw syntax error (%.a) (195ms)
    ✓ Should throw syntax error (a./();) (197ms)
    ✓ Should throw syntax error ({%: 1}) (197ms)
    ✓ Should throw syntax error (const a) (200ms)
    ✓ Should throw syntax error (delete x) (204ms)
    ✓ Should throw syntax error (function g(arguments)) (201ms)
    ✓ Should throw syntax error (function eval()) (197ms)
    ✓ Should throw syntax error (iife arguments()) (201ms)
    ✓ Should throw syntax error (catch(eval)) (200ms)
    ✓ Should throw syntax error (var eval) (199ms)
    ✓ Should throw syntax error (else) (197ms)
    ✓ Should throw syntax error (return) (196ms)
    ✓ Should throw syntax error (block-level export) (234ms)
    ✓ Should throw syntax error (block-level import) (199ms)
    ✓ Should throw syntax error (anonymous class) (199ms)
    ✓ Should throw syntax error (anonymous function) (195ms)
    ✓ Should throw syntax error (spread in sequence) (197ms)
    ✓ Should throw syntax error (for-in init) (218ms)
    ✓ Should throw syntax error (for-in var) (197ms)
```

These can be replaced with tests that run in-process instead of spawning terser as a process, making them smaller and faster too

### PR DESCRIPTION
**Pull request by fabiosantoscode**
closes #114

### CODE REVIEW

File: test/mocha/string-literal.js [hunk 1]
```diff
@@ -81,16 +81,16 @@ describe("String literals", function() {
 
     it("Should not unescape unpaired surrogates", function() {
         var code = [];
-        for (var i = 0; i <= 0xF; i++) {
+        for (var i = 0; i <= 0xF; i+=10) {
```
Comment (kzc) [inline test/mocha/string-literal.js]: @fabiosantoscode Please back out all changes to `test/mocha/string-literal.js`.

This is an extremely important test for minification integrity and has caught many bugs.

File: test/mocha/unicode.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ describe("Unicode", function() {
         it("Should not unescape unpaired surrogates", function() {
             this.timeout(20000);
             var code = [];
-            for (var i = 0; i <= 0x20001; i++) {
+            for (var i = 0; i <= 0x20001; i+=100) {
```
Comment (kzc) [inline test/mocha/unicode.js]: Likewise please back out all changes to `test/mocha/unicode.js`.

This is an extremely important test for minification integrity and has caught many bugs.

Comment (fabiosantoscode) [reply test/mocha/unicode.js]: Reverted both of these

**Comment by kzc** [comment]
@fabiosantoscode Replacing tests with 100% equivalent but faster tests is fine, such as moving syntax errors to compress tests. But please resist the urge to remove or reduce the coverage of existing regression tests. If there's a bug due to `lib/` source refactoring or user contributed PR and the test has been compromised then it's not useful.

**Comment by fabiosantoscode** [comment]
Got it

**Pull request merged**